src/mercury_engine_data_structures/formats/bmsld.py

```diff
@@ -34,125 +45,171 @@
     ),
 )
 
+ExtraActors = Struct(
+    "group" / StrId,
+    "actors" / make_vector(Struct("name" / StrId)),
+)
+
+
 BMSLD = Struct(
-    _magic=Const(b"MSLD"),
-    version=VersionAdapter("1.20.0"),
-    unk1=Int32ul,
-    unk2=Int32ul,
-    unk3=Int32ul,
-    unk4=Int32ul,
-    objects_a=make_vector(
+    "_magic" / Const(b"MSLD"),
+    "version" / VersionAdapter("1.20.0"),
+    "unk1" / CVector3D,
+    "unk2" / Float,
+    # locations where Samus gets repositioned after a cutscene/collecting dna, etc
+    "landmarks"
+    / make_vector(
         Struct(
-            name=StrId,
-            unk1=Hex(Int32ul),
-            unk2=Hex(Int32ul),
-            unk3=Hex(Int32ul),
-            unk4=Hex(Int32ul),
-            unk5=Hex(Int32ul),
-            unk6=Hex(Int32ul),
+            "name" / StrId,
+            "position" / CVector3D,
+            "rotation" / CVector3D,
         )
     ),
-    object_b=make_vector(
+    # paths that enemies follow (could be bounds?)
+    "enemy_paths"
+    / make_vector(
         Struct(
-            name=StrId,
-            unk01=Hex(Int32ul),
-            unk02=make_vector(
-                Struct(
-                    x=Float32l,
-                    y=Float32l,
-                    z=Float32l,
-                )
-            ),
+            "name" / StrId,
+            "unk01" / Hex(Int32ul),
+            "coordinates" / make_vector(CVector3D),
         )
     ),
-    objects_c=make_dict(CollisionObject),
-    objects_d=make_dict(CollisionObject),
-    objects_e=make_vector(
+    # areas of influence for enemies
+    "logic_shapes" / make_dict(CollisionObject),
+    # areas for spawngroups
+    "spawn_groups" / make_dict(CollisionObject),
+    # boss camera data
+    "boss_cameras"
+    / make_vector(
         Struct(
-            name=StrId,
-            unk01=StrId,
-            unk02=Hex(Int32ul),
-            unk03=Hex(Int32ul),
-            unk04=Hex(Int32ul),
-            unk05=Hex(Int32ul),
-            unk06=Hex(Int32ul),
-            unk07=Hex(Int32ul),
-            unk08=Hex(Int32ul),
-            unk09=Float,
-            unk10=Float,
-            unk11=Hex(Int32ul),
-            unk13=StrId,
-            unk14=Hex(Int32ul),
+            "name" / StrId,
+            "unk01" / StrId,
+            "unk02" / Hex(Int32ul),
+            "unk03" / Hex(Int32ul),
+            "unk04" / Hex(Int32ul),
+            "unk05" / Hex(Int32ul),
+            "unk06" / Hex(Int32ul),
+            "unk07" / Hex(Int32ul),
+            "unk08" / Hex(Int32ul),
+            "position" / CVector3D,
+            "type" / StrId,
+            "unk14" / Hex(Int32ul),
         )
     ),
-    actors=make_dict(ProperActor)[18],
-    sub_areas=make_vector(
+    # layers for actors
+    "actor_layers" / make_dict(ProperActor)[18],
+    # collision_cameras and groups
+    "sub_areas"
+    / make_vector(
         Struct(
-            name=StrId,
-            names=make_vector(StrId),
+            "name" / StrId,
```

Comment:
is there ever more than one subarea with the same name? if not, we should use `make_dict` here

Reply:
There aren't repeats, no. The only issue with changing this is that the functions for actor groups need to be updated to match.

Reply:
then update them